packages/block-editor/src/components/inserter/quick-inserter.js

```diff
@@ -48,27 +49,13 @@ export default function QuickInserter( {
 		destinationRootClientId
 	);
 
-	const {
-		setInserterIsOpened,
-		insertionIndex,
-		prioritizePatterns,
-	} = useSelect(
+	const { setInserterIsOpened, insertionIndex } = useSelect(
 		( select ) => {
-			const { getSettings, getBlockIndex, getBlockCount } = select(
-				blockEditorStore
-			);
+			const { getSettings } = select( blockEditorStore );
 			const settings = getSettings();
-			const index = getBlockIndex( clientId );
-			const blockCount = getBlockCount();
 
 			return {
 				setInserterIsOpened: settings.__experimentalSetIsInserterOpened,
-				prioritizePatterns:
-					settings.__experimentalPreferPatternsOnRoot &&
-					! rootClientId &&
-					index > 0 &&
-					( index < blockCount || blockCount === 0 ),
-				insertionIndex: index === -1 ? blockCount : index,
```

Comment:
I think we still need `insertionIndex ` the value is used in the `onBrowseAll` callback, and we can remove `[ clientId, rootClientId ]` deps they aren't used in the selector.

Reply:
Nice catch, the issue was fixed!